test/CMakeLists.txt

```diff
@@ -40,11 +40,10 @@ include( ${MODULE_ROOT_DIR}/fleetprovisioningFilePaths.cmake )
 add_library( coverity_analysis
              ${FLEET_PROVISIONING_SOURCES} )
 
-# Build Fleet Provisioning library without requiring config file.
-target_compile_definitions( coverity_analysis PUBLIC FLEET_PROVISIONING_DO_NOT_USE_CUSTOM_CONFIG=1 )
-
-# Fleet Provisioning public include path.
-target_include_directories( coverity_analysis PUBLIC ${FLEET_PROVISIONING_INCLUDE_PUBLIC_DIRS} )
+# Fleet Provisioning public include path and test config file.
+target_include_directories( coverity_analysis PUBLIC
```

Comment:
We pass the flag to be able to:
1. Have a mechanism for testing that the library can be built with the `DO_NOT_USE_CUSTOM_CONFIG` flag
2. To allow Coverity builds in CI to use this target for static analysis without having to provide a special config file

Reply:
1. That should likely be a separate check, as we do not want to ignore all the code in macro invocations for our testing. For example, here we had it such that the library did not build with logging macros not blank, however it passed all our tests, which is not acceptable.
2. Coverity builds in CI should be checking the logging macros so that we do not miss misra violations in the logging macro parameters. Also it just runs the cmake so will work just fine with having the cmake include the config file to enable logging.

Reply:
I have added a default config check

Reply:
> For example, here we had it such that the library did not build with logging macros not blank, however it passed all our tests, which is not acceptable.

I'm not saying that logging macros should not be enabled for tests. We have separated the test target builds from the `coverity_analysis` builds for the reasons I stated. 

I'm fine with the logging macros being included in the Coverity analysis though.